wallet/createtx.go

```diff
@@ -1444,87 +1457,93 @@ func (w *Wallet) purchaseTickets(ctx context.Context, op errors.Op, n NetworkBac
 			return err
 		})
 		if err != nil {
-			return ticketHashes, errors.E(op, err)
+			return nil, errors.E(op, err)
 		}
 
-		// Generate the ticket msgTx and sign it.
+		// Generate the ticket msgTx and sign it if DontSignTx is false.
 		ticket, err := makeTicket(w.chainParams, eopPool, eop, addrVote,
 			addrSubsidy, int64(ticketPrice), poolAddress)
 		if err != nil {
-			return ticketHashes, err
+			return nil, err
 		}
-		var forSigning []udb.Credit
-		if eopPool != nil {
-			eopPoolCredit := udb.Credit{
-				OutPoint:     *eopPool.op,
+		if !req.DontSignTx {
```

Comment:
Please use an early return here so that all of this code doesn't need to be indented more.  Keeps the patch smaller and easier to review.

Reply:
Discussed in matrix, and because this is inside a loop, a continue statement will be used here instead.